### PR TITLE
@types/node: Add missing methods to Console, plus docs

### PR DESCRIPTION
**Pull request by thw0rted**
Please fill in this template.

- [X] Use a meaningful title for the pull request. Include the name of the package modified.
- [X] Test the change in your own code. (Compile and run.)
- [ ] Add or edit tests to reflect the change. (Run with `npm test`.)
- [X] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [X] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [X] Run `npm run lint package-name` (or `tsc` if no `tslint.json` is present).

Select one of these and delete the others:

If changing an existing definition:
- [X] Provide a URL to documentation or source code which provides context for the suggested changes
[node 9.X](https://nodejs.org/docs/latest-v9.x/api/console.html), [node 10.X](https://nodejs.org/api/console.html)
- [X] Increase the version number in the header if appropriate. (N/A)

### CODE REVIEW

**Comment by thw0rted** [comment]
If there's a better way to handle "this method only works with `--inspect`" I am open to suggestions.

File: types/node/index.d.ts [hunk 1]
```diff
@@ -32,17 +32,101 @@
...
     debug(message?: any, ...optionalParams: any[]): void;
+    /**
+     * Uses {@link util.inspect()} on `obj` and prints the resulting string to `stdout`.
+     * This function bypasses any custom `inspect()` function defined on `obj`.
+     */
+    dir(obj: any, options?: NodeJS.InspectOptions): void;
+    /**
+     * The `console.dirxml()` function is an alias for {@link console.log()}.
+     */
+    dirxml(...data: any[]): void;
```
Comment (Flarna) [inline types/node/index.d.ts]: Why is there a difference in the type definition of `dirxml` to `log` if one is just an alias of the other?

Comment (thw0rted) [reply types/node/index.d.ts]: This is based on the Node docs, which say that the method signature is `...data <any>`, and that it "calls console.log() passing it the arguments received".  The [official spec](https://console.spec.whatwg.org/#assert) agrees with Node's definition, even though [the MDN page](https://developer.mozilla.org/en-US/docs/Web/API/Console/dirxml) says it's "non-standard" and only takes exactly one argument.  Of course, logically `foo(...everything: any[])` works identically to `foo(first?: any, ...others: any[])` (from the caller's perspective) so it's kind of academic.

All that to say: I copied it from other similar methods (like `debug`) and probably should have used the original docs from Node,

> This method calls console.log() passing it the arguments received. Please note that this method does not produce any XML formatting.

Comment (Flarna) [reply types/node/index.d.ts]: I found following in NodeJS Source:
```Console.prototype.debug = Console.prototype.log;
Console.prototype.info = Console.prototype.log;
Console.prototype.dirxml = Console.prototype.log;```
So seems there is really no difference between them.

But you are right, for the user it's don't care which variant is used here in the typings.

**Review by Flarna — APPROVED** [review]

**Comment by thw0rted** [comment]
Looks like your Travis is jacked, the failures are unrelated to these changes.  Who can address?

**Comment by thw0rted** [comment]
Hey @Flarna and others watching, I have a DT / node question and figured it probably wasn't worth opening a new issue / PR until I understand why things are the way they are.  For a long time, Node's typings have included [a `NodeBuffer` interface](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/types/node/v9/index.d.ts#L825), marked `@deprecated`, which the actual `Buffer` interface extends without adding anything.  I don't think `NodeBuffer` is a real interface in Node proper and isn't referenced anywhere in the current (v9 / v10) docs, even under Deprecated APIs.  Is this just a placeholder for some reason?  Is there any reason not to just rename NodeBuffer to Buffer directly and remove the deprecation decorator?

If my guess is right, I'll make a new PR (maybe after this one merges successfully?) that does this.

**Pull request merged**